### PR TITLE
Added optional params subject_text and body_text…

### PR DESCRIPTION
**Pull request by knifhen**
…  that can be used instead of reading subject and body from files

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,7 +47,9 @@ Sending an email message:
 #### Parameters
 
 * `to`: _(mandatory)_ A list of recipients as strings.
+* `subject_text`: _(optional)_ Plain text email subject, instead of `subject`.
```
Comment (mdomke) [inline README.md]: Could you make clear that either the `subject_text` _or_ the `subject` option have to be supplied. The same applies for the `body` and the `body_text`.

File: opt/resource/out [hunk 2]
```diff
@@ -67,22 +67,30 @@ def get_mime_subtype(payload):
...
 
 def get_subject(src, payload, template_vars):
-    fname = get_value_from_payload(payload, 'params', 'subject')
-    with open(os.path.join(src, fname)) as fp:
-        return Template(fp.read()).render(template_vars)
+    subject_text = get_value_from_payload(payload, 'params', 'subject_text', optional=True)
+    if not subject_text:
+        fname = get_value_from_payload(payload, 'params', 'subject')
+        with open(os.path.join(src, fname)) as fp:
+            return Template(fp.read()).render(template_vars)
```
Comment (mdomke) [inline opt/resource/out]: I would rather use a variable `subject` for both config values `'subject_text'` and `'subject'` and assign its value accordingly. Additionally I think that the template variables should be applied in any case.

``` python
    subject = get_value_from_payload(payload, 'params', 'subject_text', optional=True)
    if not subject:
        fname = get_value_from_payload(payload, 'params', 'subject')
        with open(os.path.join(src, fname)) as fp:
            subject = fp.read()
    return Template(subject).render(template_vars)
```

File: opt/resource/out [hunk 3]
```diff
@@ -67,22 +67,30 @@ def get_mime_subtype(payload):
...
+    else:
+        return subject_text
 
 
 def get_message(src, payload, template_vars):
-    fname = get_value_from_payload(payload, 'params', 'body')
-    inline = get_value_from_payload(payload, 'params', 'inline_css', default=True, optional=True)
-    mime_subtype = get_mime_subtype(payload)
+    body_text = get_value_from_payload(payload, 'params', 'body_text', optional=True)
+    if not body_text:
```
Comment (mdomke) [inline opt/resource/out]: As described above. The body should be read either from the constant text in `'body_text'` or from the file `'body'`. All other operations (e.g: css inlining, template variable application) should be applied regardless of the source of the body-text.

``` python
body = get_value_from_payload(payload, 'params', 'body_text', optional=True)
if not body:
    fname = get_value_from_payload(payload, 'params', 'body')
    with open(os.path.join(src, fname)) as fp:
        message = fp.read()
message = Template(message).render(template_vars)
...
if mime_subtype == 'html' and inline:
    ...
return message
```

**Review by mdomke — CHANGES_REQUESTED** [review]
Hi @knifhen! Thanks for your pull-request! I appreciate your contribution very much! Nevertheless I think that some things should be handled a little different from the way you implemented it. It would be great if you could adjust your pull-request accordingly or give some feedback on why you think it should be as it is.

**Comment by knifhen** [comment]
Hi @mdomke, I tried to make the pull request more to your specification. That was some good points you made about the change. Hope this works better!

Regards Andreas Knifh

**Review by mdomke — APPROVED** [review]

**Pull request merged**